### PR TITLE
changed translatex to make cards look cleaner when arranged

### PR DESCRIPTION
**Pull request by martanman**
before:
chrome_VHkS6A6Lxh
chrome_NhJJUxkjDT
chrome_pCSBXm5kZI
image

after:
chrome_ZIqDMxqoP6
chrome_AirhCpsZ9e
chrome_qck16VbNv9
image

### CODE REVIEW

File: client/src/interfaces/PropTypes.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export interface DroppedClassProps {
   cardWidth: number;
   clashIndex: number;
   clashColour: string;
+  recWidth: number; // width of a grid cell
```
Comment (munjoonteo) [inline client/src/interfaces/PropTypes.ts]: can u call this `cellWidth` lol

File: client/src/components/timetable/DroppedClasses.tsx [hunk 2]
```diff
@@ -195,6 +195,24 @@ const DroppedClasses: React.FC<DroppedClassesProps> = ({ assignedColors, handleS
     return defaultValues;
   };
 
+  // handles getting width of a cell in the grid
+  const myRef = React.useRef<HTMLDivElement>(null);
+  const [recWidth, setRecWidth] = useState(0);
+  useLayoutEffect(() => {
+    function updateRecWidth() {
```
Comment (munjoonteo) [inline client/src/components/timetable/DroppedClasses.tsx]: no declaring functions with the `function` keyword

File: client/src/components/timetable/DroppedClass.tsx [hunk 3]
```diff
@@ -197,17 +176,12 @@ const StyledCourseClassInner = styled(Card, {
   background-color: ${({ backgroundColor }) => backgroundColor};
   color: white;
   font-size: 0.9rem;
-  border-radius: ${({ isSquareEdges }) => (isSquareEdges ? '0px' : '7px')};
```
Comment (munjoonteo) [inline client/src/components/timetable/DroppedClass.tsx:200]: rest in pieces square corners. did u mean to delete this?

**Review by munjoonteo — CHANGES_REQUESTED** [review]

File: client/src/components/timetable/DroppedClasses.tsx (hunk 2 above)
Comment (martanman) [reply client/src/components/timetable/DroppedClasses.tsx]: when I copy code from stack overflow and I'm scared it won't work unless I copy it exactly

File: client/src/components/timetable/DroppedClass.tsx (hunk 3 above)
Comment (martanman) [reply client/src/components/timetable/DroppedClass.tsx:200]: yup, https://mui.com/material-ui/api/paper/ paper has prop `square`.

**Review by munjoonteo — APPROVED** [review]

File: client/src/components/timetable/DroppedClass.tsx [hunk 4]
```diff
@@ -27,46 +27,19 @@ import { getClassMargin, rowHeight } from './TimetableLayout';
...
-
-    // The reason for multiplying by widthRatio is because translating a card x% will only translate
-    // a card of half the width x/2% and so on. Namely, widthRatio is compensating for the shorter distance
-    // that the card is being translated by due to its smaller width.
-    return `${((cardData.time.day - 1) * 100 + clashIndex * width) * widthRatio - microOffset}%`;
+    return `${(cellWidth + 1) * (cardData.time.day - 1) + clashIndex * (cellWidth / widthRatio)}px`;
   }
 
   // This cardData is for an unscheduled class, i.e. it belongs in the inventory
   // 5 / devicePixelRatio refers to the five timetable borders between Monday and the Unscheduled column
```
Comment (munjoonteo) [inline client/src/components/timetable/DroppedClass.tsx]: Can you update this comment please

File: client/src/components/timetable/DroppedClass.tsx [hunk 5]
```diff
@@ -27,46 +27,19 @@ import { getClassMargin, rowHeight } from './TimetableLayout';
...
-    // (cardData.time.day - 1) * 100 moves the class card to the correct day column
-    // as all cards are spawned at the Monday 9am cell
-    // Then, clashIndex * width shifts the card to the right so that all clashing classes
-    // are (basically) evenly distributed in a particular timetable cell
-
-    // The reason for multiplying by widthRatio is because translating a card x% will only translate
-    // a card of half the width x/2% and so on. Namely, widthRatio is compensating for the shorter distance
-    // that the card is being translated by due to its smaller width.
-    return `${((cardData.time.day - 1) * 100 + clashIndex * width) * widthRatio - microOffset}%`;
+    return `${(cellWidth + 1) * (cardData.time.day - 1) + clashIndex * (cellWidth / widthRatio)}px`;
```
Comment (munjoonteo) [inline client/src/components/timetable/DroppedClass.tsx]: Some explanation here would be nice

**Review by munjoonteo — CHANGES_REQUESTED** [review]

File: client/src/components/timetable/DroppedClass.tsx (hunk 5 above)
Comment (martanman) [reply client/src/components/timetable/DroppedClass.tsx]: thought it was kinda self explanatory but sure

**Review by munjoonteo — APPROVED** [review]

**Pull request merged**